src/Dispatcher.js

```diff
@@ -11,17 +14,49 @@ export default class ReduxDispatcher {
     redux: PropTypes.object.isRequired
   };
 
-  constructor(props) {
-    this.dispatcher = createDispatcher();
-    this.dispatcher.receiveStores(props.stores);
+  getChildContext() {
+    return { redux: this };
   }
 
-  componentWillReceiveProps(nextProps) {
-    this.dispatcher.receiveStores(nextProps.stores);
+  constructor(props, context) {
+    super(props, context);
+
+    this.subscriptions = [];
+    this.emitChange = this.emitChange.bind(this);
+    this.dispatch = this.dispatch.bind(this);
+
+    const initialAtom = dispatch(props.store, undefined, {});
+    this.setAtom(initialAtom);
   }
 
-  getChildContext() {
-    return { redux: this.dispatcher };
+  dispatch(action) {
+    const nextAtom = dispatch(this.props.store, this.atom, action);
+    this.setAtom(nextAtom);
+  }
+
+  setAtom(atom) {
+    this.atom = atom;
+    if (this.state) {
+      this.setState({ atom });
```

Comment:
Now that I think of it.. Do we even need to keep it in `state`? I don't think so! `Injector`s subscribe to the updates via `context.subscribe()`.

Now, `Injector`s, too receive the full `atom`, but we can easily implement more efficient / fine-grained `Injector`s in the future that only check their part of the tree.